src/events.rs

```diff
@@ -398,6 +400,9 @@ async fn run_event_collector(
     mut receiver: mpsc::Receiver<Event>,
 ) {
     while let Some(ev) = receiver.recv().await {
+        // IMPORTANT: DO NOT REMOVE DEBUG LINE COMMENT BELLOW
+        println!("{:?}", ev);
```

Comment:
I think it might make sense to use a logging library like [`tracing`](https://crates.io/crates/tracing) for this kind of debug stuff. It adds another step of enabling a formatter when you actually want to see the debug info, but it also makes it opt-in for library consumers.

Reply:
This is interesting, is this library used for logging as well? I was keeping an eye on `slog`

Reply:
Uhmm, not sure if I want to go with the extra deps for this functionality, I committed the code with that line uncommented, but in theory that should only be uncommented if the test-suite hangs